internal/provider/security/security_certificate_data_source.go

```diff
@@ -0,0 +1,172 @@
+				Optional:            true,
+			},
+			"common_name": schema.StringAttribute{
+				MarkdownDescription: "Common name of the certificate.",
+				Optional:            true,
+			},
+			"svm_name": schema.StringAttribute{
+				MarkdownDescription: "SVM name in which the certificate is installed.",
+				Optional:            true,
+			},
+			"scope": schema.StringAttribute{
+				MarkdownDescription: "Set to 'svm' for certificates installed in a SVM. Otherwise, set to 'cluster'.",
+				Computed:            true,
+			},
+			"type": schema.StringAttribute{
+				MarkdownDescription: "Type of Certificate.",
+				Computed:            true,
+			},
+			"serial_number": schema.StringAttribute{
+				MarkdownDescription: "Serial number of certificate.",
+				Computed:            true,
+			},
+			"ca": schema.StringAttribute{
+				MarkdownDescription: "Certificate authority.",
+				Computed:            true,
+			},
+			"hash_function": schema.StringAttribute{
+				MarkdownDescription: "Hashing function.",
+				Computed:            true,
+			},
+			"key_size": schema.Int64Attribute{
+				MarkdownDescription: "Key size of the certificate in bits.",
+				Computed:            true,
+			},
+			"expiry_time": schema.StringAttribute{
+				MarkdownDescription: "Certificate expiration time, in ISO 8601 duration format or date and time format.",
+				Computed:            true,
+			},
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Certificate uuid.",
+				Computed:            true,
+			},
+		},
+	}
+}
+
+// Configure adds the provider configured client to the data source.
+func (d *SecurityCertificateDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+	config, ok := req.ProviderData.(connection.Config)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected Config, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+	}
+	d.config.ProviderConfig = config
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (d *SecurityCertificateDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data SecurityCertificateDataSourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	errorHandler := utils.NewErrorHandler(ctx, &resp.Diagnostics)
+	// we need to defer setting the client until we can read the connection profile name
+	client, err := connection.GetRestClient(errorHandler, d.config, data.CxProfileName)
+	if err != nil {
+		// error reporting done inside NewClient
+		return
+	}
+
+	restInfo, err := interfaces.GetSecurityCertificate(errorHandler, *client, data.Name.ValueString(), data.CommonName.ValueString())
```

Comment:
What if both Name and CommonName are not provided (since they both are optional)? Need to handle this case.

Reply:
`internal/provider/security/security_certificate.go` currently has the below logic (L38-L44).
Please let me know if this error reporting/handling is not correct!
        ```
if name != "" {
		query.Set("name", name)
	} else if common_name != "" {
		query.Set("common_name", common_name)
	} else {
		return nil, errorHandler.MakeAndLogError("Error: 'name' or 'common_name' are required parameters.")
	}
```

Reply:
The users may use "terraform validate" (or users use "terraform apply" without "yes") to check if the parameters: data passed to an attribute matches its desired data type and if all mandatory attributes are present for the resource or data source file. Checking the parameters in validation stage will be better. Check this out and see if it can be applied to your case. https://github.com/hashicorp/terraform-plugin-framework-validators/blob/main/stringvalidator/at_least_one_of_example_test.go

Reply:
I've added a couple of validations inside `ConfigValidators` function for the data source.